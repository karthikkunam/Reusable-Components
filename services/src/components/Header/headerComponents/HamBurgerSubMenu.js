import React, { useState } from 'react';
import PropTypes from 'prop-types';

const HamBurgerSubMenu = props => {
  const [selectedUrl, setSelectedUrl] = useState(props.defaultSubNav);
  const { subMenuItems } = props;

  const onSubMenuClick = href => {
    const { onMenuChange } = props;
    onMenuChange && onMenuChange(href);
    setSelectedUrl(href);
  };

  return (
    <div className="body">
      {subMenuItems &&
        subMenuItems.length > 0 &&
        subMenuItems.map(subMenu => {
          return (
            <div key={subMenu.header}>
              <div className="header">{subMenu.header}</div>
              <ul className="list">
                {subMenu.subMenuList &&
                  subMenu.subMenuList.length > 0 &&
                  subMenu.subMenuList.map(item => {
                    return (
                      <li key={item.title} className="list-item">
                        <a
                          href={item.url}
                          className={`list-item-link menu-open level-1-link ${
                            item.url === selectedUrl ? 'active' : ''
                          }`}
                          onClick={() => {
                            onSubMenuClick(item.url);
                          }}>
                          {item.title}
                        </a>
                      </li>
                    );
                  })}
              </ul>
            </div>
          );
        })}
      {/* <div className="footer">
        <div className="logout"></div>
        <div className="version-text">
          <span>Inventory Version</span>
          <span>1.1.2.3.5.6</span>
        </div>
      </div> */}
    </div>
  );
};
HamBurgerSubMenu.defaultProps = {
  defaultSubNav: null,
};
HamBurgerSubMenu.propTypes = {
  subMenuItems: PropTypes.arrayOf({}),
  onMenuChange: PropTypes.func,
  defaultSubNav: PropTypes.string,
};

export default HamBurgerSubMenu;
