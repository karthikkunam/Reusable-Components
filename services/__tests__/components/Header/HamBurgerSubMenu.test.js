import React from 'react';
import HamBurgerSubMenu from '../../../src/components/Header/headerComponents/HamBurgerSubMenu';
import { shallow } from 'enzyme';

const props = {
  subMenuItems: [
    {
      header: 'Header 1',
      subMenuList: [
        {
          title: 'page 1 Title',
          url: 'page1',
        },
      ],
    },
    {
      header: 'Header 2',
      subMenuList: [
        {
          title: 'page 4 Title',
          url: 'page4',
        },
      ],
    },
  ],
  onMenuChange: jest.fn(),
};

describe('HamBurgerSubMenu Component', () => {
  it('renders correctly', () => {
    const component = shallow(<HamBurgerSubMenu {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders navigateURL onClick', () => {
    const component = shallow(<HamBurgerSubMenu {...props} />);
    expect(component).toMatchSnapshot();
    component
      .find('ul')
      .at(0)
      .find('li')
      .at(0)
      .find('a')
      .simulate('click', {
        onSubMenuClick: () => {},
      });
  });
});
