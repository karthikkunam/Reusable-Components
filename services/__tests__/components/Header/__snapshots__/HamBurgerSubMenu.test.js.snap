// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamBurgerSubMenu Component renders correctly 1`] = `
<div
  className="body"
>
  <div
    key="Header 1"
  >
    <div
      className="header"
    >
      Header 1
    </div>
    <ul
      className="list"
    >
      <li
        className="list-item"
        key="page 1 Title"
      >
        <a
          className="list-item-link menu-open "
          href="page1"
          onClick={[Function]}
        >
          <label>
            page 1 Title
          </label>
        </a>
      </li>
    </ul>
  </div>
  <div
    key="Header 2"
  >
    <div
      className="header"
    >
      Header 2
    </div>
    <ul
      className="list"
    >
      <li
        className="list-item"
        key="page 4 Title"
      >
        <a
          className="list-item-link menu-open "
          href="page4"
          onClick={[Function]}
        >
          <label>
            page 4 Title
          </label>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`HamBurgerSubMenu Component renders navigateURL onClick 1`] = `
<div
  className="body"
>
  <div
    key="Header 1"
  >
    <div
      className="header"
    >
      Header 1
    </div>
    <ul
      className="list"
    >
      <li
        className="list-item"
        key="page 1 Title"
      >
        <a
          className="list-item-link menu-open "
          href="page1"
          onClick={[Function]}
        >
          <label>
            page 1 Title
          </label>
        </a>
      </li>
    </ul>
  </div>
  <div
    key="Header 2"
  >
    <div
      className="header"
    >
      Header 2
    </div>
    <ul
      className="list"
    >
      <li
        className="list-item"
        key="page 4 Title"
      >
        <a
          className="list-item-link menu-open "
          href="page4"
          onClick={[Function]}
        >
          <label>
            page 4 Title
          </label>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
